### PR TITLE
Auto refresh live every 15 seconds

### PR DESCRIPTION
**Pull request by sphinxrave**

### CODE REVIEW

File: apps/client-api/services/CacheService.js [hunk 1]
```diff
@@ -14,9 +14,23 @@ class CacheService {
     return {};
   }
 
+  async getStringFromCache(key) {
```
Comment (IAmVisco) [inline apps/client-api/services/CacheService.js:17]: But why

File: apps/client-api/routes/v1/util/live.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+const { Op } = require('sequelize');
```
Comment (IAmVisco) [inline apps/client-api/routes/v1/util/live.js:1]: I see why you moved it but file path kinda looks weird. Maybe a service or something.

**Review by IAmVisco — CHANGES_REQUESTED** [review]

File: apps/crawler-youtube/tasks/live-cache-update.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    const lookback_hours = 0;
+    const hide_channel_desc = 1;
+    const channel_simple = undefined;
+
+    const cacheKey = (channel_id || max_upcoming_hours || lookback_hours || hide_channel_desc || channel_simple)
+      ? `live-${channel_id}-${max_upcoming_hours}-${lookback_hours}-${hide_channel_desc}-${channel_simple}` : 'live';
+
+    await computeLive(cacheKey, channel_id, max_upcoming_hours, lookback_hours, hide_channel_desc, channel_simple);
+  })();
+};
```
[the remark this answers is not in the document]
Comment (dragonjet) [reply apps/crawler-youtube/tasks/live-cache-update.js:28]: he probably wants to move execution to next process without waiting for the first block's `await computeLive`. could probably function it and `Promise.all` both calls, but I'm also fine with current, doesn't hurt.

File: apps/client-api/services/CacheService.js (hunk 1 above)
Comment (sphinxrave) [reply apps/client-api/services/CacheService.js:17]: grabbing this live JSON and deserializing it is completely useless.

File: apps/client-api/routes/v1/live.js [hunk 4]
```diff
@@ -21,80 +18,15 @@ router.get('/', asyncMiddleware(async (req, res) => {
...
-          },
-        },
-      ],
-      where: {
-        live_end: { [Op.gte]: nowMoment.clone().subtract(lookback || VIDEOS_PAST_HOURS, 'hour').toISOString() },
-      },
-    });
-    results.ended = pastVideos;
+  if (cache) {
+    return res.end(cache);
```
[the remark this answers is not in the document]
Comment (sphinxrave) [reply apps/client-api/routes/v1/live.js:25]: res.json you have to give it a json object (which it stringifies) instead i give it a string and handle the header myself. faster this way.

File: apps/client-api/routes/v1/util/live.js (hunk 2 above)
Comment (sphinxrave) [reply apps/client-api/routes/v1/util/live.js:1]: let's just keep it here for now, some refactoring work required for v2 anyway

**Pull request merged**